rfcs/rfc-0000-pronouns.md

```diff
@@ -15,56 +15,70 @@ Pronouns have been requested by the community and are high up on the RFC list.
 
 # Guide-level explanation
 
-Currently users display their pronouns in their Profile content (bio) but this provices a cleaner solution to the problem. The aim of this change is to ensure that pronouns for a user can be easily seen and clearly displayed seperately to a users bio.
+Currently users display their pronouns in their Profile content (bio) but this provides a cleaner solution to the problem. The aim of this change is to ensure that pronouns for a user can be easily seen and clearly displayed separately to a users bio. Additionally this change will create local database localised pronoun validation.
 
 # Reference-level explanation
 
-This change will modify:
-- core/models/users
-- core/db/util/bridge
-- quark/models/users
-- core/db/models/users/ops/mongo
-  And possibly more
+The field `pronouns` will be added to the main `User` object, as well as the `PartialUser` and `FieldsUser` objects. (In practise this means that the pronouns field will be fetched for the `fetch_user` API route, and pronouns can be edited or removed from the `edit_user` API route.
 
-For the backend, pronoun data will be stored as such.
+The pronoun field (on a `User` will be an optional array of strings. 
+
+### Data representation in Rustlang
 ```rust
-    // User's displayed pronouns
-        #[cfg_attr(feature = "validator", validate(length(min = 1, max = 5)))]
-        #[serde(skip_serializing_if = "Option::is_none")]
-        pub pronouns: Option<Vec<String>>,
+{
+pronouns: Option<Vec<String>>
+}
 ```
 
-Having pronouns on Revolt will be optional and this data will be editable from the Profile section of user settings on Revolt's main client. One of the major changes this could bring is pronoun specific language for bots interacting with users.
+### Pronouns stored in JSON
+```json
+{
+	"pronouns": ["she", "her"]
+}
+```
+For no pronouns the object will not be serialised 
+```json
 
-#### Input
-```text
-?/ whois info:roles @ToastXC
 ```
-#### Response
-```text
-She has no roles!
+
+Additionally, there will be database changes for pronouns to be fully implemented. The `Collection` shall contain an entry of every supported language (represented by an abbreviation) with an array of every valid pronoun for the language. This collection must also be accessible as an API route.
+
+```json
+{
+  "language": "ENG",
+  "pronouns": ["she", "her", ...]
```

Comment:
This addition raises another question of if people will be able to mix and match pronouns since they will technically be separated in the backend. For example, if someone's pronouns are he/she or even more like he/she/they or more, will that person be able to combine them? Or will that have to be its own preset pronoun set sent in by the community.

Reply:
i assumed that was the whole point of having an array of words, as opposed to using singular strings like `"he/him"`, `she/her`, etc

Reply:
With the current plans users will be able to use a group of individual pronouns, if someone's pronouns are `she/her` then that will be stored as ["she", "her"].